### PR TITLE
DDF-3856 Fix automatic injection of wild card

### PR DESCRIPTION
**Pull request by gjvera**
#### What does this PR do?
Fixes automatic injection of wild card. Wild card will now only be injected if there are no other parameters 
#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
@Schachte @mojogitoverhere 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here. If you don't know who to ask, you can request reviews in https://groups.google.com/forum/#!forum/ddf-developers .
(please choose ONLY two committers from below, delete the rest)
@bdeining
@jlcsmith
#### How should this be tested? (List steps with links to updated documentation)
1. Run a quick build 
2. once installed, run  <log:set DEBUG org.apache.cxf>
3. Configure an opensearch source 
4. Run a search with no parameters 
5. Ensure the output still has the wild card parameter in the outputted URL
6. Add parameters to the search, Verify wild card is not in the outputted URL and the query is still performed correctly. 
#### What are the relevant tickets?
DDF-3856
[](https://codice.atlassian.net/browse/)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/opensearch/catalog-opensearch-source/src/main/java/org/codice/ddf/opensearch/source/OpenSearchSource.java [hunk 1]
```diff
@@ -383,8 +383,9 @@ private SourceResponse doOpenSearchQuery(
       WebClient restWebClient)
       throws UnsupportedQueryException {
     // All queries must have at least a search phrase to be valid
-    searchPhraseMap.putIfAbsent(OpenSearchConstants.SEARCH_TERMS, "*");
-
+    if (searchPhraseMap.isEmpty()) {
```
Comment (millerw8) [inline catalog/opensearch/catalog-opensearch-source/src/main/java/org/codice/ddf/opensearch/source/OpenSearchSource.java]: We really only want to do this if everything is empty... no searchPhrase, no temporalSearch, no spatialSearch.
We should be able to represent this:
image
like this: services/catalog/query?lat=20&lon=20&radius=500
I believe this should be legal

Comment (gjvera) [reply catalog/opensearch/catalog-opensearch-source/src/main/java/org/codice/ddf/opensearch/source/OpenSearchSource.java]: I believe so. As long as `parameters` is not empty, the query should be executed just fine. However, with the new fix, it fails some of the opensearch source tests in `OpenSearchSpec.groovy`. Some of these tests require that `q=*` is a parameter for the query. I can alter the tests to reflect the change we want to make, but it's up to you @millerw8

**Comment by bdeining** [comment]
build now

**Comment by GabrielFabian** [comment]
Performed OS query with wildcard, outbound address:

```https://localhost:8993/services/gsrmap/catalog/query?sortKeys=DateRevised,,0&start=1&startPage=1&count=100&dedup=1&source=GSR&timeout=10000&dn=admin&mapFilter=CONTAINS%7B*%7D%3AAnyText&src=local```


Performed OS query without wildcard, queried on alt. id., outbound address:


```https://localhost:8993/services/gsrmap/catalog/query?sortKeys=DateRevised,,0&start=1&startPage=1&count=100&dedup=1&source=GSR&timeout=10000&dn=admin&mapFilter=CONTAINS%7Babcde%7D%3AAlternateIdentifier%2FValue&src=local```

Hero build successful! LGTM 🎉

**Pull request merged**